lib/pbench/test/unit/server/test_schema.py

```diff
@@ -0,0 +1,153 @@
+import dateutil
+import pytest
+from typing import Callable
+
+from pbench.server.api.auth import Auth, UnknownUser
+from pbench.server.api.resources.query_apis import (
+    ParamType,
+    ConversionError,
+    InvalidRequestPayload,
+    MissingParameters,
+    Parameter,
+    Schema,
+)
+
+
+class TestParamType:
+    """
+    Tests on the ParamType enum
+    """
+
+    def test_enum(self):
+        assert (
+            len(ParamType.__members__) == 4
+        ), "Number of ParamType ENUM values has changed; confirm test coverage!"
+        for n, t in ParamType.__members__.items():
+            assert str(t) == t.friendly.upper()
+            assert isinstance(t.convert, Callable)
+
+    @pytest.mark.parametrize(
+        "test",
+        (
+            (ParamType.STRING, "x", "x"),
+            (ParamType.JSON, {"key": "value"}, {"key": "value"}),
+            (ParamType.DATE, "2021-06-29", dateutil.parser.parse("2021-06-29")),
+            (ParamType.USER, "drb", "drb"),
+        ),
+    )
+    def test_successful_conversions(self, test, monkeypatch):
+        def ok(user: str) -> str:
+            return user
+
+        monkeypatch.setattr(Auth, "validate_user", ok)
```

Comment:
It turns out, we have a PyText "fixture" for [this](https://github.com/distributed-system-analysis/pbench/blob/bd5d0657bfedcc651e3d9a90d810341f701aac07/lib/pbench/test/unit/server/conftest.py#L110)....